tcms/testruns/static/testruns/js/get.js

```diff
@@ -86,9 +84,50 @@ function renderProgressBars(positiveCount, negativeCount, allCount) {
 
 function renderCountPerStatusList(statusCount) {
     for (var status in statusCount) {
-        const status_id = statusCount[status].id;
+        const statusId = statusCount[status].id;
 
-        $(`#count-for-status-${status_id}`).attr('href', `?status_id=${status_id}`);
-        $(`#count-for-status-${status_id}`).text(statusCount[status].count);
+        $(`#count-for-status-${statusId}`).attr('href', `?status_id=${statusId}`).text(statusCount[status].count);
     }
 }
+
+function renderTestExecutions(testRunId, executionStatuses) {
+    const container = $('#test-executions-container')
+    const testExecutionRowTemplate = $('#test-execution-row')[0].content
+
+    jsonRPC('TestExecution.filter', { 'run_id': testRunId }, testExecutions => {
```

Comment:
This API is called twice with the same argument which potentially could result in double loading thousands of repeated records.

1st call is in `drawPercentBar()` and the second is here. 

Although the first function really cares only about the statuses it receives the full data set from the server.

Call `TestExecution.filter` at the document.ready() level and pass the data around to the functions which need it.